lib/hello_web/controllers/comment_controller.ex

```diff
@@ -0,0 +1,73 @@
+defmodule HelloWeb.CommentController do
+  use HelloWeb, :controller
+
+  alias Hello.Topic.Comment
+
+  def action(conn, _) do
+    # conn |> IO.inspect(label: "COnn in action")
+    topic = Comment.get_topic!(conn.params["topic_id"])
+    # topic |> IO.inspect(label: "topic from conn params t id")
+    args = [conn, conn.params, topic]
+    apply(__MODULE__, action_name(conn), args)
+  end
+
+  def list_comments(topic_id) do
+    Comment.list_topic_comments(topic_id)
+  end
+
+  def create_changeset(topic_id) do
```

Comment:
i would just call the  `Comment.changeset` function directly wherever you are calling this one